R/package.R

```diff
@@ -178,18 +178,39 @@ pkgdown_config_path <- function(path) {
   )
 }
 
-read_meta <- function(path) {
+read_meta <- function(path, call = caller_env()) {
   path <- pkgdown_config_path(path)
 
   if (is.null(path)) {
     yaml <- list()
   } else {
-    yaml <- yaml::yaml.load_file(path) %||% list()
+    yaml <- cli_yaml_load(path, call = call) %||% list()
   }
-
   yaml
 }
 
+# Wrapper around yaml::yaml.load_file()
+cli_yaml_load <- function(path, call = caller_env()) {
+  # Tweak the original message to put the location of the error at the end
+  withCallingHandlers(
+    yaml::yaml.load_file(
+      path,
+      error.label = NULL # suppress path from message.
+    ),
+    error = function(e) {
+      yaml_err <- conditionMessage(e)
+      # Rethrow cli-styled error!
+      cli::cli_abort(c(
+          "x" = "Could not parse the config file.",
+          "!" = yaml_err,
+          "i" = "Edit {.path {path}} to fix the problem."
+        ),
+        call = call
+      )
+    }
+  )
```

Comment:
Is there a reason not to use `parent = e` here? (Rather than the custom message wrapping?) If you do that, I think the code will be simple enough that you no longer need the wrapper function.

Reply:
I think it is better now!